pkg/apis/monitoring/v1/prometheus_types.go

```diff
@@ -692,6 +692,10 @@ type CommonPrometheusFields struct {
 	// +listType=map
 	// +listMapKey=name
 	ScrapeClasses []ScrapeClass `json:"scrapeClasses,omitempty"`
+
+	// +kubebuilder:default="Endpoints"
+	// +kubebuilder:validation:Enum=Endpoints;EndpointSlice
+	ServiceDiscoveryRole string `json:"serviceDiscoveryRole,omitempty"`
```

Comment:
Wouldn't that be a breaking change anyway?

Reply:
I think what Simon meant from the suggested change is:

- If users don't set ServiceDiscoveryRole, or set it as Endpoints, we use Endpoints
- Only if users set it as EndpointSlice, we use EndpointSlice

This way I don't think it would break current users' setup? Since we're already using Endpoints.

Reply:
Yes, but I also understood that we want to change to EndpoitSlice by default in the future 😬. I was wondering how safe this change would be if we ever happen to make it.

But that's a problem for a future PR, going forward with Endpoints by default sounds the correct approach here :)

Reply:
I think how safe that change would be depends very much on how popular EndpoinSlice would be. I researched this a bit (for ServiceMonitor support in GSoC proposal). There're questions around when/whether Kubernetes can deprecate Endpoints, since EndpointSlice has already proved to perform better.

If some far day Endpoints are not used anymore, I think we can safely make EndpointSlice as the default.